internal/service/currency-service.go

```diff
@@ -100,14 +108,13 @@ func getApiUrl() string {
 
 // UpdateCurrencyRates is used to update #currencyInfo by calling 3rd party API.
 // In this case #currencyInfo is a cache value of API response for currency USD.
-func (s *CurrencyService) UpdateCurrencyRates() {
+func (s *CurrencyService) UpdateCurrencyRates() error {
 	log.Println("Start updating currency rates")
 
 	// Get list of 3rd party values.
 	currencyRates, err := getCurrencyRateFromApi()
 	if err != nil {
-		log.Panic("Failed to update currency rates: ", err)
-		return
+		return err
```

Comment:
You can wrap error here

https://preslav.me/2023/04/14/golang-error-handling-is-a-form-of-storytelling/
```suggestion
		return fmt.Errorf("getting currency rates: %w", err)
```